staramr/blast/results/BlastHitPartitions.py

```diff
@@ -55,9 +51,9 @@ def _find_parition(self, hit):
         return None
 
     def _hit_in_parition(self, hit, partition):
```

Comment:
Thanks. Is type hinting available in Python 3.5?  I remember you mentioned it was 3.6+.  Right now I build a conda package for Python 3.5, so I'm not sure if it's worth it to get rid of the 3.5 package for type hinting (though I assume most people using it don't care what Python they are using, since it's through conda anyways).

Reply:
Yup! You should be able to use them in Python 3.5 like in my example code. I use them in `bio_hansel` and it builds for either 3.5 or 3.6. You will need to import from the `typing` module though. I've also used type hinting in Python 2.7 with a bit more work (you need to specify them in a comment instead and I think `import mypy`).

Reply:
Thanks. I've done this in 46bd872192aa80d1aced65db73bdf2ad0062e514 (and verified it works in Python 3.5).

But, I was having trouble with the `Dict` part.  I was getting an error:

```
TypeError: Parameters to generic types must be types. Got slice(<class 'str'>, typing.Union[str, int, typing.List[staramr.blast.results.AMRHitHSP.AMRHitHSP]],.
```

I couldn't quite figure it out, but it does work if I just change to `Dict`.  Would you happen to know what's going on?

Reply:
Ah... I think you need it to be `Optional[Dict]` since it can be `None`. The `Dict[str, Union[str, int, List[...]]` is just specifying the expected types of the key and values of the dict. Hope that helps!

Reply:
Thanks.  That wasn't it but I figured it out in 98faf1f.  You have to use `Dict[key, value]` instead of `Dict[key: value]`.

Reply:
Yeah it takes a bit of getting used to. You still might want to wrap the `Dict[...]` with `Optional[Dict[...]]` since the return value can be `None`.

Reply:
Must have forgot about that. Done in 9ea2835